plugins/modules/wikiReader.py

```diff
@@ -0,0 +1,138 @@
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights Osmose project 2024                                        ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+
+# This module file contains functions to read MediaWiki markup tables, templates, lists, ...
+
+import wikitextparser
+
+# Get a list of lists containing all cells of a table.
+# Parameters:
+#   wikitext (str) - the text of a wikipedia page
+#   tab_index (int) - the index of the table (if there's multiple tables on the wiki)
+#   keep_markup (bool) - if False, everything (except Templates) will be converted to plain text
+#   skip_headers (bool) - if True, header rows are removed. Assumes all headers are on top
+# Returns:
+#   The cell contents, specified as a list in a list.
+#   The outer list is the rows, the inner list are the cells in that row
+# Throws:
+#   If the table at the specified index isn't found
+def read_wiki_table(wikitext, tab_index = 0, keep_markup = False, skip_headers = True):
```

Comment:
As it is a in modules. Maybe type the function rather than explaining the typing.

Reply:
Done. It seems that typing is only checked by mypy if the calling function is also typed though

Reply:
> Done. It seems that typing is only checked by mypy if the calling function is also typed though

Thank. Yes, but you can also force to check it.

Reply:
You mean with `check_untyped_defs = True`?
Because that gives a lot of errors `Found 928 errors in 183 files (checked 432 source files)`